apps/worker/tasks/upload_finisher.py

```diff
@@ -347,6 +347,12 @@ def run_impl(
 
         except SoftTimeLimitExceeded:
             log.warning("run_impl: soft time limit exceeded")
+            # Clean up orphaned state so future finishers can proceed
+            state.mark_uploads_as_merged(upload_ids)
```

Comment:
I think this is wrong, this will mark all uploads as `merged` correct? But if we hit a `SoftTimeLimitExceeded`, I'm guessing this is not true